vault-agent/variables.tf

```diff
@@ -0,0 +1,29 @@
+variable "install_dependencies" {
+  description = "Whether to install/download application dependencies."
+  type        = bool
+}
+
+variable "vault_agent" {
+  description = "Configuration for Vault Agent"
+  type = object({
+    enabled             = bool
+    auth_method         = object({
+      type   = string
+      config = object({
+        role_id   = string # Content of the role ID file
+        secret_id = string # Content of the secret ID file
+      })
+    })
+    vault_address       = string
+    vault_ca_cert       = string # Content of the CA certificate file
+    templates           = list(object({
+      source_path      = string
+      destination_path = string
+      service_name     = string
+      secret_path      = string
+      secret_key       = string
+    }))
+    agent_config        = string
+    release_version     = string
```

Comment:
I feel ambiguous about that. Clearly, we are validating with a version and cannot guarantee it will work for all version.

I'd either fixate the version completely (with a local variable) or otherwise at least put a default value with the version we validated with.